pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -210,70 +212,70 @@ type KubeControllerManagerConfiguration struct {
 	// GarbageCollectorControllerConfiguration holds configuration for
 	// GarbageCollectorController related features.
-	GarbageCollectorController GarbageCollectorControllerConfiguration
+	GarbageCollectorController GarbageCollectorControllerConfiguration `json:"garbageCollectorController"`
 	// HPAControllerConfiguration holds configuration for HPAController related features.
-	HPAController HPAControllerConfiguration
+	HPAController HPAControllerConfiguration `json:"hpaController"`
 	// JobControllerConfiguration holds configuration for JobController related features.
-	JobController JobControllerConfiguration
+	JobController JobControllerConfiguration `json:"jobController"`
 	// NamespaceControllerConfiguration holds configuration for NamespaceController
 	// related features.
-	NamespaceController NamespaceControllerConfiguration
+	NamespaceController NamespaceControllerConfiguration `json:"namespaceController"`
 	// NodeIpamControllerConfiguration holds configuration for NodeIpamController
 	// related features.
-	NodeIpamController NodeIpamControllerConfiguration
+	NodeIpamController NodeIpamControllerConfiguration `json:"nodeIpamController"`
 	// NodeLifecycleControllerConfiguration holds configuration for
 	// NodeLifecycleController related features.
-	NodeLifecycleController NodeLifecycleControllerConfiguration
+	NodeLifecycleController NodeLifecycleControllerConfiguration `json:"nodeLifecycleController"`
 	// PersistentVolumeBinderControllerConfiguration holds configuration for
 	// PersistentVolumeBinderController related features.
-	PersistentVolumeBinderController PersistentVolumeBinderControllerConfiguration
+	PersistentVolumeBinderController PersistentVolumeBinderControllerConfiguration `json:"persistentVolumeBinderController"`
 	// PodGCControllerConfiguration holds configuration for PodGCController
 	// related features.
-	PodGCController PodGCControllerConfiguration
+	PodGCController PodGCControllerConfiguration `json:"podGCController"`
 	// ReplicaSetControllerConfiguration holds configuration for ReplicaSet related features.
-	ReplicaSetController ReplicaSetControllerConfiguration
+	ReplicaSetController ReplicaSetControllerConfiguration `json:"replicaSetController"`
 	// ReplicationControllerConfiguration holds configuration for
 	// ReplicationController related features.
-	ReplicationController ReplicationControllerConfiguration
+	ReplicationController ReplicationControllerConfiguration `json:"replicationController"`
 	// ResourceQuotaControllerConfiguration holds configuration for
 	// ResourceQuotaController related features.
-	ResourceQuotaController ResourceQuotaControllerConfiguration
+	ResourceQuotaController ResourceQuotaControllerConfiguration `json:"resourceQuotaController"`
 	// SAControllerConfiguration holds configuration for ServiceAccountController
 	// related features.
-	SAController SAControllerConfiguration
+	SAController SAControllerConfiguration `json:"saController"`
```

Comment:
`sa` or `serviceAccount`?

Reply:
probably `serviceAccount` after a refactor, but this just makes things consistent without renaming fields as a stop-gap. I want to keep this PR as small as possible and do field rename refactor in https://github.com/kubernetes/kubernetes/pull/66993

Reply:
Is the assumption that none of these serialized fields are currently used in supported config files? It's very clear this struct should not be serialized, since it is missing serialization tags. Once you add them in this PR, it's a lot harder to tell whether someone has ever written out config using this struct they expect to be able to read back in. Can we correct the field names first before adding tags for serialization?